### PR TITLE
Emojis `HashMap` will be updated, Serenity will retry connecting, cleaned up code

### PR DESCRIPTION
**Pull request by Lakelezz**
When a guild updates its Emojis, Serenity will `clone()` the new `HashMap` instead of just `extend()`.

If `connect()` returns an `Err()`, it will retry connecting.

Cleaned up `help_command.rs`.

### CODE REVIEW

File: src/gateway/shard.rs [hunk 1]
```diff
@@ -1024,6 +1025,19 @@ fn build_gateway_url(base: &str) -> Result<Url> {
         .map_err(|_| Error::Gateway(GatewayError::BuildingUrl))
 }
 
+/// Tries to connect and upon failure, retries.
+fn connecting() -> WsClient {
+    loop {
+        match connect(&String::new()) {
```
Comment (acdenisSK) [inline src/gateway/shard.rs]: replace `&String::new()` with just `""`

File: src/gateway/shard.rs [hunk 2]
```diff
@@ -1024,6 +1025,21 @@ fn build_gateway_url(base: &str) -> Result<Url> {
         .map_err(|_| Error::Gateway(GatewayError::BuildingUrl))
 }
 
+/// Tries to connect and upon failure, retries.
+fn connecting() -> WsClient {
+    let waiting_time = 120;
+
+    loop {
+        match connect("") {
```
Comment (zeyla) [inline src/gateway/shard.rs]: This won't work, because a base URL to the gateway is needed.

File: src/gateway/shard.rs [hunk 3]
```diff
@@ -1024,6 +1025,21 @@ fn build_gateway_url(base: &str) -> Result<Url> {
         .map_err(|_| Error::Gateway(GatewayError::BuildingUrl))
 }
 
+/// Tries to connect and upon failure, retries.
+fn connecting() -> WsClient {
+    let waiting_time = 120;
```
Comment (zeyla) [inline src/gateway/shard.rs]: Let's make this smaller, like 30.

File: src/gateway/shard.rs [hunk 4]
```diff
@@ -1024,6 +1025,21 @@ fn build_gateway_url(base: &str) -> Result<Url> {
...
+
+    loop {
+        match connect("") {
+            Ok(client) => return client,
+            Err(_) => {
```
Comment (zeyla) [inline src/gateway/shard.rs]: Might be nice to log what this error is.

File: src/model/event.rs [hunk 5]
```diff
@@ -404,7 +404,7 @@ impl CacheUpdate for GuildEmojisUpdateEvent {
 
     fn update(&mut self, cache: &mut Cache) -> Option<()> {
         cache.guilds.get_mut(&self.guild_id).map(|guild| {
-            guild.with_mut(|g| g.emojis.extend(self.emojis.clone()))
+            guild.with_mut(|g| g.emojis = self.emojis.clone())
```
Comment (zeyla) [inline src/model/event.rs]: I don't like `(|foo| foo = bar)` syntax, can we make this:
```rust
guild.with_mut(|g| {
    g.emojis = ...;
});
```

Additionally, please use `HashMap::clone_from`.

**Comment by zeyla** [comment]
Thanks

**Pull request merged**